core/src/costs.rs

```diff
@@ -39,501 +38,461 @@ impl<P: PageNumber> CostPerPage<P> {
         }
     }
 
-    /// Calculate cost for `pages`.
-    pub fn calc(&self, pages: P) -> u64 {
-        self.cost.saturating_mul(pages.raw() as u64)
-    }
-
-    /// Cost for one page.
+    /// Cost for one.
     pub const fn one(&self) -> u64 {
         self.cost
     }
+}
 
-    /// Returns another [CostPerPage] with increased `cost` to `other.cost`.
-    pub const fn saturating_add(&self, other: Self) -> Self {
-        Self::new(self.cost.saturating_add(other.cost))
+impl<T: Into<u32>> CostPer<T> {
+    /// Calculate cost for `num` amount of `T`.
+    pub fn calc(&self, num: T) -> u64 {
+        self.cost.saturating_mul(Into::<u32>::into(num).into())
     }
 }
 
-impl<P: PageNumber> Debug for CostPerPage<P> {
+impl<T> Debug for CostPer<T> {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
         f.write_fmt(format_args!("{}", &self.cost))
     }
 }
 
-impl<P: PageNumber> From<u64> for CostPerPage<P> {
+impl<T> From<u64> for CostPer<T> {
     fn from(cost: u64) -> Self {
-        CostPerPage::new(cost)
+        CostPer::new(cost)
     }
 }
 
-impl<P: PageNumber> From<CostPerPage<P>> for u64 {
-    fn from(value: CostPerPage<P>) -> Self {
+impl<T> From<CostPer<T>> for u64 {
+    fn from(value: CostPer<T>) -> Self {
         value.cost
     }
 }
 
-impl<P: PageNumber> Default for CostPerPage<P> {
+impl<T> Default for CostPer<T> {
     fn default() -> Self {
-        CostPerPage::new(0)
+        CostPer::new(0)
     }
 }
 
-/// Describes the weight for each imported function that a program is allowed to call.
-#[derive(Clone, Encode, Decode, PartialEq, Eq, Default)]
-pub struct HostFnWeights {
-    /// Weight of calling `alloc`.
-    pub alloc: u64,
+/// Some actions or calls amount.
+#[derive(Debug, Default, Clone, Copy, derive_more::From, derive_more::Into)]
+pub struct Calls(u32);
 
-    /// Weight per allocated page for `alloc`.
-    pub alloc_per_page: u64,
+/// Bytes amount.
+#[derive(Debug, Default, Clone, Copy, derive_more::From, derive_more::Into)]
+pub struct Bytes(u32);
```

Comment:
I think such types should be renamed because their names are too common and pollute namespace easily

Reply:
Renamed